File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task93_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>

struct Record {
    std::string key;
    std::string value;
};

bool compare(const Record &a, const Record &b) {
    return a.key < b.key;
}

int main() {
    std::ifstream file("input.txt");
    if (!file) {
        std::cerr << "Error opening file" << std::endl;
        return 1;
    }
    std::vector<Record> records;
    std::string line;
    while (std::getline(file, line)) {
        size_t pos = line.find('-');
        Record record;
        record.key = line.substr(0, pos);
        record.value = line.substr(pos + 1);
        records.push_back(record);
    }
    std::sort(records.begin(), records.end(), compare);
    for (const auto &record : records) {
        std::cout << record.key << "-" << record.value << std::endl;
    }
    file.close();
    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task93_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        try (BufferedReader br = new BufferedReader(new FileReader("input.txt"))) {
            List<String[]> records = new ArrayList<>();
            String line;
            while ((line = br.readLine()) != null) {
                String[] record = line.split("-");
                records.add(record);
            }
            Collections.sort(records, new Comparator<String[]>() {
                @Override
                public int compare(String[] o1, String[] o2) {
                    return o1[0].compareTo(o2[0]);
                }
            });
            for (String[] record : records) {
                System.out.println(record[0] + "-" + record[1]);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}